rita/src/rita_client/exit_manager/mod.rs

```diff
@@ -29,15 +106,43 @@ impl Handler<Tick> for ExitManager {
     type Result = Result<(), Error>;
 
     fn handle(&mut self, _: Tick, _ctx: &mut Context<Self>) -> Self::Result {
-        if SETTING.exit_client_is_set() && SETTING.exit_client_details_is_set() {
-            TrafficWatcher::from_registry().do_send(Watch(
-                Identity {
-                    mesh_ip: SETTING.get_exit_client().exit_ip,
-                    wg_public_key: SETTING.get_exit_client_details().wg_public_key.clone(),
-                    eth_address: SETTING.get_exit_client_details().eth_address,
+        let exit_server = {
+            SETTING
+                .get_exit_client()
+                .get_current_exit()
+                .map(|c| c.clone())
+        };
+
+        // code that connects to the current exit server
+        if let Some(exit) = exit_server {
+            if let Some(ref details) = exit.details {
+                linux_setup_exit_tunnel().expect("failure setting up exit tunnel");
```

Comment:
this may be safe name just setup_exit_tunnel nothing changes with this operation on opewnrt.